vars/buildDependencies.groovy

```diff
@@ -9,9 +9,38 @@ def call(buildInformation) {
    echo "Building dependencies for job ${env.JOB_NAME}."
    
    def dependencyBuild
+   def dependencyDir
    
    buildInformation.dependencies.each{dependency->
       try {
+         dependencyDir = "${dependency}_DEP_DIR"
+
+         // Allow Jenkinsfile to override which dependencies to use
+         if(env."$dependencyDir") {
+            String nodeLabel = ''
+            if (buildInformation.nodeLabel?.trim()) {
+               nodeLabel = buildInformation.nodeLabel
+            }
+
+            def dependencyFound = false
+            node(nodeLabel) {
+               stage("Dependency Validation") {
+                  def desiredDependency = env."$dependencyDir"
+                  if(fileExists(desiredDependency)) {
+                     echo "Using dependency $desiredDependency."
+                     dependencyFound = true
+                  }
+                  else {
+                     echo "$desiredDependency does not exist. Rebuilding $dependency."
+                  }
+               }
+            }
+
+            if(dependencyFound) {
+               return
```

Comment:
Is there a limitation preventing us from returning in the `if(fileExists(...` case directly? I am assuming we can't return properly from in the stage.

Reply:
Correct. I originally had the return directly in the stage, but then, it just returns from the stage and executes the rest of the function. This actually returns from the function.